### PR TITLE
For #28028, update desktop to use core login framework

### PR DESCRIPTION
**Pull request by jfboismenu-adsk**
- Uses tank.platform.interactive_authentication module for managing login and logout.
- Uses tank.util.login.get_current_user to get the Human User that is logged in.
  - get_current_user code introduces a regression where the Shotgun account associated to the os user
   will be always be returned insead of the one associated with the login credentials. Once the core
   is updated so that get_current_user returns the user currently authenticated as, this regression will
   disappear.

### CODE REVIEW

**Comment by jfboismenu-adsk** [comment]
This should be reviewed in tandem with

File: python/tk_desktop/desktop_window.py [hunk 1]
```diff
@@ -504,9 +503,11 @@ def show_update_project_config(self):
         self.update_project_config_widget.show()
         self.project_overlay.hide()
 
+    def _get_current_user(self):
+        return login.get_current_user(sgtk.platform.current_engine().tank)
```
Comment (manneohrstrom) [inline python/tk_desktop/desktop_window.py]: Is this inside the engine? Could you use `sgtk.platform.current_bundle()`? That's the standard way to get the app/engine object from code inside the app/engine (but only works if you have used `import_module`).

Comment (manneohrstrom) [inline python/tk_desktop/desktop_window.py]: `get_current_user()` tries to match the operating system user right? Should we add a comment here that this is going to be fixed soon? Or is it ok this way?

File: python/tk_desktop/project_commands_model.py [hunk 2]
```diff
@@ -267,7 +264,7 @@ def _handle_command_triggered(self, item, command_name=None, button_name=None,
             "event_type": self.APP_LAUNCH_EVENT_TYPE,
             "project": self.__project,
             "meta": {"name": command_name, "group": group_name},
-            "user": login,
+            "user": login.get_current_user(engine.tank),
```
Comment (manneohrstrom) [inline python/tk_desktop/project_commands_model.py]: sgtk please!

File: python/tk_desktop/project_model.py [hunk 3]
```diff
@@ -218,10 +217,7 @@ def __init__(self, parent, overlay_parent_widget):
         """ Constructor """
         ShotgunModel.__init__(self, parent, download_thumbs=True)
 
-        # override the default connection to use one that is tied to the current login
-        connection = ShotgunLogin.get_instance_for_namespace("tk-desktop").get_connection()
-
-        self.set_shotgun_connection(connection)
+        self.set_shotgun_connection(shotgun.create_sg_connection())
```
Comment (manneohrstrom) [inline python/tk_desktop/project_model.py]: does this need its own instance? Or can it grab a shared instance?

**Comment by manneohrstrom** [comment]
CR comments: 
I wasn't sure about the use of the current login - are we making that chance in anticipation of that coming change? Also the use of current engine perhaps feels little non-standard (but should work fine!) - wondering if current-bundle may be better?

File: python/tk_desktop/desktop_window.py (hunk 1 above)
Comment (jfboismenu-adsk) [inline python/tk_desktop/desktop_window.py]: I attached a comment to the commit to that effect, but it should be in the code as well. Will do.

Comment (jfboismenu-adsk) [inline python/tk_desktop/desktop_window.py]: This code is part of the engine, but I don't understand what's the difference between calling the two? current_engine has the merit of being explicit about what I think I am getting back.

File: python/tk_desktop/project_model.py (hunk 3 above)
Comment (jfboismenu-adsk) [inline python/tk_desktop/project_model.py]: It should probably reuse a thread-local one. I'll grab the sgtk.shotgun instance.

**Pull request closed without merging**